mobile/src/main/java/ch/epfl/sdp/mobile/ui/speech_recognition/SpeechRecognitionFilter.kt

```diff
@@ -0,0 +1,14 @@
+package ch.epfl.sdp.mobile.ui.speech_recognition
+
+/** Filtering interface for speech recognizer results */
+interface SpeechRecognitionFilter {
+  /**
+   * @param speeches list of recognized speeches
+   * @param confidencesScores assigned scores to each recognized speech
+   * @return filtered results
+   */
+  fun filterWords(
+      speeches: List<String>,
+      confidencesScores: List<Double>,
+  ): String
```

Comment:
What happens if the input lists are empty ? This should maybe return a `String?`

Reply:
Resolved in DMs. We concluded that it's more convenient and less error prone to use nullable string as a return type instead of assuming empty strings for unsuccessful filtering.